src/main/java/nextstep/subway/applicaion/LineCommandService.java

```diff
@@ -3,51 +3,80 @@
 import nextstep.subway.applicaion.dto.LineResponse;
-import nextstep.subway.applicaion.dto.StationResponse;
+import nextstep.subway.applicaion.dto.SectionRequest;
 import nextstep.subway.domain.Line;
 import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.Station;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import java.util.List;
-import java.util.NoSuchElementException;
-
 @Service
 @RequiredArgsConstructor
-@Transactional(readOnly = true)
+@Transactional(readOnly = false)
 public class LineCommandService {
 
     private static final String LINE_NOTFOUND_MESSAGE = "해당 id의 지하철 노선이 존재하지 않습니다.";
 
     private final StationQueryService stationQueryService;
+
+    private final StationCommandService stationCommandService;
+
+    private final SectionCommandService sectionCommandService;
+
+    private final SectionStationCommandService sectionStationCommandService;
+
     private final LineRepository lineRepository;
 
-    @Transactional
     public LineResponse saveLine(LineRequest lineRequest) {
-        Line line = lineRepository.save(new Line(lineRequest));
+        Station upStation = stationQueryService.findById(lineRequest.getUpStationId());
+        Station downStation = stationQueryService.findById(lineRequest.getDownStationId());
+
+        Section section = sectionCommandService.saveSection(new Section(lineRequest.getDistance()));
+        sectionStationCommandService.saveSectionStation(section, upStation, downStation);
+        Line line = lineRepository.save(new Line(lineRequest, section));
+
         return createLineResponse(line);
     }
 
-    @Transactional
     public void updateLine(Long id, LineRequest lineRequest) {
         Line line = findLineOrElseThrow(id);
-        lineRepository.save(lineRequest.toLine(line));
+        Station upStation = stationQueryService.findById(lineRequest.getUpStationId());
+        Station downStation = stationQueryService.findById(lineRequest.getDownStationId());
+
+        line.update(lineRequest, upStation, downStation);
     }
 
-    @Transactional
     public void deleteLineById(Long id) {
         findLineOrElseThrow(id);
         lineRepository.deleteById(id);
     }
 
+    public void addSection(Long id, SectionRequest sectionRequest) {
+        Line line = findLineOrElseThrow(id);
+        line.checkSectionRulesOrThrow(sectionRequest.getUpStationId(), sectionRequest.getDownStationId());
+        Station upStation = stationQueryService.findById(sectionRequest.getUpStationId());
+        Station downStation = stationQueryService.findById(sectionRequest.getDownStationId());
+
+        Section section = sectionCommandService.saveSection(new Section(sectionRequest.getDistance()));
+        sectionStationCommandService.saveSectionStation(section, upStation, downStation);
+        line.addSection(section);
+    }
```

Comment:
이렇게 서비스 계층에서 모든 비즈니스 로직이 있는 것을 `트랜잭션 스크립트 패턴` 이라 합니다. 

하나의 서비스 계층에서 모든 로직을 처리해야 하니 복잡해지고 여러 의존성이 얽히고 설키게 되는 문제가 있어요. 구간정보가 어디에 종속되는 개념인지와 일급 컬렉션을 활용해보면 다음과 같이 될 수 있지 않을까요?

```suggestion
    public void addSection(Long id, SectionRequest sectionRequest) {
        Line line = findLineOrElseThrow(id);
        Station upStation = stationQueryService.findById(sectionRequest.getUpStationId());
        Station downStation = stationQueryService.findById(sectionRequest.getDownStationId());
        
        line.addSection(new Section(line, upStation, downStation, sectionRequest.getDistance());
    }
```